eis_toolkit/prediction/weights_of_evidence.py

```diff
@@ -0,0 +1,255 @@
+    classes = np.unique(evidence)
+    return {cls: _calculate_metrics_for_class(deposits, evidence == cls) for cls in classes}
+
+
+def _cumulative_weights(deposits: np.ndarray, evidence: np.ndarray, ascending: bool = True) -> dict:
+    """Calculate cumulative weights (ascending or descending) for each class."""
+    classes = sorted(np.unique(evidence), reverse=not ascending)
+    cumulative_classes = [classes[: i + 1] for i in range(len(classes))]
+    return {
+        cls[i]: _calculate_metrics_for_class(deposits, np.isin(evidence, cls))
+        for i, cls in enumerate(cumulative_classes)
+    }
+
+
+def _reclassify_by_studentized_contrast(df: pd.DataFrame, studentized_contrast_threshold: Number) -> None:
+    """Create generalized classes based on the studentized contrast threhsold value."""
+    index = df.idxmax()["Contrast"]
+
+    if df.loc[index, "Studentized contrast"] < studentized_contrast_threshold or index == len(df.index) - 1:
+        raise exceptions.ClassificationFailedException(
+            "Failed to create generalized classes with given studentized contrast treshold ({})".format(
+                df.loc[index, "Studentized contrast"]
+            )
+        )
+
+    df["Generalized class"] = 1
+    for i in range(0, index + 1):
+        df.loc[i, "Generalized class"] = 2
+
+
+def _calculate_generalized_weights(df: pd.DataFrame, deposits) -> None:
+    """
+    Calculate generalized weights.
+
+    Implementation for generalized weights that uses the SAME logic than the original implementation.
+    """
+    total_deposits = np.sum(deposits == 1)
+    total_no_deposits = deposits.size - total_deposits
+
+    # Class 2
+    class_2_max_index = df.idxmax()["Contrast"]
+    class_2_count = df.loc[class_2_max_index, "Pixel count"]
+    class_2_point_count = df.loc[class_2_max_index, "Deposit count"]
+
+    class_2_w_gen = np.log(class_2_point_count / total_deposits) - np.log(
+        (class_2_count - class_2_point_count) / total_no_deposits
+    )
+    clas_2_s_wpls_gen = np.sqrt((1 / class_2_point_count) + (1 / (class_2_count - class_2_point_count)))
+
+    df["Generalized W+"] = round(class_2_w_gen, 4)
+    df["Generalized S_W+"] = round(clas_2_s_wpls_gen, 4)
+
+    # Class 1
+    class_1_count = df.loc[len(df.index) - 1, "Pixel count"] - class_2_count
+    class_1_point_count = df.loc[len(df.index) - 1, "Deposit count"] - class_2_point_count
+
+    class_1_w_gen = np.log(class_1_point_count / total_deposits) - np.log(
+        (class_1_count - class_1_point_count) / total_no_deposits
+    )
+    clas_1_s_wpls_gen = np.sqrt((1 / class_1_point_count) + (1 / (class_1_count - class_1_point_count)))
+    df.loc[df["Generalized class"] == 1, "Generalized W+"] = round(class_1_w_gen, 4)
+    df.loc[df["Generalized class"] == 1, "Generalized S_W+"] = round(clas_1_s_wpls_gen, 4)
+
+
+def _generate_rasters_from_metrics(
+    evidence: np.ndarray, df: pd.DataFrame, metrics_to_include: List[str] = ["Class", "W+", "S_W+"]
```

Comment:
```suggestion
    evidence: np.ndarray, df: pd.DataFrame, metrics_to_include: List[str] = DEFAULT_METRICS
```

I would suggest refactoring the default metrics to a module variable `DEFAULT_METRICS = ["Class", "W+", "S_W+"]` as the same list is used elsewhere as well.